lib/get-distro.js

```diff
@@ -0,0 +1,65 @@
+// External Modules
+const fs = require('fs');
+const os = require('os');
+
+// Internal Modules
+const getOS = require('./get-os');
+const log = require('./utils').log;
+
+// Constants
+const DEFAULT_OS_PATH = '/etc/os-release';
+
+// Helper in Getting Dist from GetOS
+const getDist = (callback) => {
+    if (typeof callback !== 'function') {
+        throw new Error('Must provide a function callback');
+    }
```

Comment:
you don't want to throw Errors when trying to execute an async function

in this case, you are checking if callback is actually a function, if it isn't, you should just do a no-op, sure the Error helps to identify the situation, but it doesn't help rescue you from now what is potentially a hung process

also, since this is a local function, you should be able to ensure that you are passing in the correct parameters without these safety checks -- and if you do need to ensure the params are sanitized, they should be done in the publicly exported function, not the internal private function

Reply:
Gone!